server/src/test/java/org/elasticsearch/script/ScriptContextStatsTests.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script;
+
+import org.elasticsearch.test.ESTestCase;
+
+import static org.elasticsearch.script.TimeSeriesTests.randomTimeseries;
+
+public class ScriptContextStatsTests extends ESTestCase {
+    public void testMerge() {
+        {
+            var first = randomScriptContextStats();
+            var second = randomScriptContextStats();
+
+            var e = expectThrows(AssertionError.class, () -> ScriptContextStats.merge(first, second));
```

Comment:
I think if this is important enough to test, we should make it an exception not an assertion. Assertions are not always enabled so this could slip through. What do you think?

Reply:
If I correctly understand, asserts are more to enforce that the developer is doing the correct thing (generate valid states of the system by code), and exceptions, to say like "Hey the system state is not valid due to the user input". In this case, the merge of 2 `ScriptContextStats` is only made by the developer, so that's why I used an assert :P .